UI/src/view/defaultLocationButton.js

```diff
@@ -1,23 +1,62 @@
 (function (root) {
-  root.DefaultLocationButton = function (map, container, backend) {
+  root.DefaultLocationButton = function (map, container, backend, assetTypeConfig, defaultAssetType, defaultCoordinates) {
+    var vkm = function() {
+      if(zoom <= 5){
+        backend.getMunicipalityFromCoordinates(lon, lat, function (vkmResult) {
+              var municipalityInfo = vkmResult.kunta ? vkmResult.kunta : "Suomi";
+              municipality = municipalityInfo;
+              ely = vkmResult.ely_nimi ? vkmResult.ely_nimi : "Suomi";
+            }, function () {
+              municipality = 'Suomi';
+              ely = 'Suomi';
+            }
+        );
+      }
+    };
+
+    var defaultAsset = defaultAssetType;
+    var defaultLocation = defaultCoordinates;
+    var municipality = 'Suomi';
+    var ely = 'Suomi';
+    var lon = defaultLocation.shift();
+    var lat = defaultLocation.shift();
+    var zoom = defaultLocation.shift();
+    var roles;
+    var places;
+
     var element =
       '<div class="default-location-btn-container"></div>';
 
     container.append(element);
 
 
     eventbus.on('roles:fetched', function(userInfo) {
-      if (!(_.some(userInfo.roles, function(role) {return role === "viewer"; })))
-        $('.default-location-btn-container').append('<button class="btn btn-sm btn-tertiary" id="default-location-btn">Muuta oletussijainniksi</button>');
+      if (!(_.some(userInfo.roles, function(role) {return (role === "viewer" || role === "serviceRoadMaintainer");}))) {
+          roles = userInfo.roles[0];
```

Comment:
What if the user has multiple roles? How can you be sure to always pick the first one?

Reply:
The users who can have a combination of roles are ely maintainers and service maintainers between themselves and the way a user is created with that combination will be ely first in roles[]

Reply:
Maybe it is a part of a bigger issue and needs to be cleaned up, but in production there are users with ["premium", "busStopMaintainer"].

Reply:
Ok i will rewrite this piece of code then